sensor/common/component.go

```diff
@@ -85,6 +85,15 @@ type ComplianceComponent interface {
 	ComplianceC() <-chan MessageToComplianceWithAddress
 }
 
+// CollectorComponent is a sensor component that can communicate with collector. All the messages intended for
+// collector are returned by CollectorC(). It must be started before the collector.Multiplexer or we panic.
+type CollectorComponent interface {
+	SensorComponent
+	Stopped() concurrency.ReadOnlyErrorSignal
+
+	CollectorC() chan sensor.MsgToCollector
```

Comment:
Let's make it read-only to catch potential problems early.
```suggestion
	CollectorC() <-chan sensor.MsgToCollector
```